packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -822,7 +820,12 @@ export abstract class FluidDataStoreContext
 		this._isInMemoryRoot = true;
 	}
 
-	public abstract getBaseGCDetails(): Promise<IGarbageCollectionDetailsBase>;
+	/**
+	 * @deprecated - The functionality to get base GC details has been moved to summarizer node.
+	 */
+	public async getBaseGCDetails(): Promise<IGarbageCollectionDetailsBase> {
```

Comment:
removal and deprecation usually do not go in one change. Are you better off submitting this change (with removal of functionality) in next?
If you want to go to main, I think we need to ensure that N-2 versions are not calling this API, right?

Reply:
Ideally, yes. This one is a little special (and weird). There are a few things at play here:
1. IFluidDataStoreContext is an interface interface and not public. We still have to make sure we don't break it since it crosses the container-runtime datastore-runtime boundary.
2. The change for datastore-runtime to not rely on getBaseGCDetails() already went in to internal.2.4 - https://github.com/microsoft/FluidFramework/commit/23408d9cbfe61477c25e23fbab0a28f034472a3a.
3. The SummarizerNode logic that now handles passing in the baseGCDetails to child nodes is only in container-runtime layer. Even though datastore-runtime layer creates a summarizer node, it uses the function passed in by the datastore context which in the runtime layer. So, this does not break back-compat.